src/raw/alloc.rs

```diff
@@ -47,4 +47,20 @@ mod inner {
     pub fn do_alloc<A: Allocator>(alloc: &A, layout: Layout) -> Result<NonNull<u8>, ()> {
         alloc.allocate(layout).map_err(|_| ())
     }
+
+    #[cfg(feature = "bumpalo")]
+    unsafe impl Allocator for bumpalo::Bump {
+        fn allocate(&self, layout: Layout) -> Result<NonNull<u8>, AllocError> {
```

Comment:
Just curious, I'm comparing this with the implementation I made upstream, and the signature looks wrong.
`allocator_api` signature should be `NonNull<[u8]>`. That makes me require the `nonnull_slice_from_raw_parts` feature.

It feels like I'm missing something here.

Reply:
That's just an internal `Allocator` trait, it is completely private and can't be implemented outside this crate. As such, it doesn't *have* to match the standard library `Allocator` trait.

Reply:
ha! I missed that, thank you!